### DISCUSSION TITLE
What is the purpose of the GDS file in the PDK?

### DISCUSSION THREAD

**Discussion by 706697916** [The-OpenROAD-Project/OpenROAD-flow-scripts#1629] [General]
The platform config file of any PDK in ORFS needs to export the tech_file, standard_cell_file, lib_file and GDS file. It is shown below in the picture. I wonder what is the GDS file of the PDK and why it is needed in ORFS.
![385cab9d313c576c787008a46dafeab](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/70256434/f161de7b-1fbd-406e-8e9e-965b0da4dad7)

**Comment by rovinski** [comment]
GDS ([Graphic Design System](https://en.wikipedia.org/wiki/GDSII)) is the file format which contains the actual layout polygons for a cell. LEF is an abstract format and only contains some layers. Usually a LEF view of a cell (e.g. an AND gate) will contain only the metal layers. At the end of the flow, the LEF view of the cell is replaced by the GDS view which will contain all of the layers (including, importantly, the transistors and other substrate layers).

ORFS will generate a GDS view of your chip. The GDS is what needs to be submitted to the foundry to make your chip.

**Reply by 706697916** [reply]
But why the ORFS needs the GDS file as an input of PDK? I think the GDS file is the output

**Reply by 706697916** [reply]
![10aed45f41fcabe519016dcf3e19b39](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/70256434/da9dd4d0-801f-4b44-b55e-5d5099f064c7)
As you can see, it is the input.

**Reply by maliberty** [reply]
All the standard cells and macros need a gds as the LEF abstract doesn't have enough detail to produce a gds.

**Reply by rovinski** [reply]
To put it more simply:
You provide ORFS your RTL. This RTL is translated into gates. OpenROAD uses the LEF view of the gates, which only has the metal information. Once OpenROAD is done, it needs to replace the LEF view with the GDS view of the cells (so that it has the transistor layers, etc). OpenROAD merges all of this info together into a single, final GDS for your chip.

**Reply by 706697916** [reply]
Thanks for your reply.
